### ISSUE TITLE
Unexpected crash when input is Chinese and file contains html formatting

### ISSUE THREAD

**Issue by masterrite** [wereturtle/ghostwriter#737]
**IMPORTANT NOTICE**:  Please do NOT use the issue tracker for new feature requests.  Only bug reports or troubleshooting requests are being accepted at this time.  For details, please see the [contributing guidelines](https://github.com/wereturtle/ghostwriter/blob/master/CONTRIBUTING.md).

**Platform (please complete the following information):**
 - Windows 10 21H2 19044.1566

**ghostwriter installation (please complete the following information):**
- Ghostwriter 2.1.1 full install

**Describe the bug**
Ghostwriter crashes in a few seconds after typing in a file that previously had html formating and Chinese characters. Exception code 0xc0000005 and fault offset 0x000000000001b490 in event viewer.

I have not had this happen in previous ghostwriter versions. I'm not sure if this is a bug in Ghostwriter or confined to vs-Ghostwriter. I'm submitting this there, too

**To Reproduce**
1. Create a file with html formatting, eg. <center> ** </center>
2. enter Chinese texts
3. save and close
4. reopen the file and try to enter some more Chinese characters
5. the Ghostwriter instance should crash in a few seconds

**Expected behavior**
It shouldn't crash

**Screenshots**
Not applicable

**Additional context**
Here's the event log from event viewer:
```
Faulting application name: ghostwriter.exe, version: 0.0.0.0, time stamp: 0x61c67bf6
Faulting module name: ghostwriter.exe, version: 0.0.0.0, time stamp: 0x61c67bf6
Exception code: 0xc0000005
Fault offset: 0x000000000001b490
Faulting process id: 0x4f0
Faulting application start time: 0x01d827ada07fb4a5
Faulting application path: C:\Program Files\ghostwriter\ghostwriter.exe
Faulting module path: C:\Program Files\ghostwriter\ghostwriter.exe
Report Id: c5ff3163-437d-4b26-993d-fff0b67a1bbf
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by wereturtle** [comment]
I think this issue *may* be fixed in the new 2.1.2 version.  Please recompile and try again and let me know if the issue persists.

Also, if it does persist, please include a copy of some specific text with which I can replicate the issue.

Thanks!

**Comment by masterrite** [comment]
Thanks for the reply! I'll try it out when the windows version is released and report back.

**Comment by masterrite** [comment]
I don't think 2.1.2 release is out yet

**Comment by wereturtle** [comment]
Yes I hadn't put out official release notes in GitHub yet.  It was only uploaded/tagged.  It's fully out now on the GitHub release page.  Unfortunately I don't handle the Windows builds anymore so you may have to wait a while.

**Comment by wereturtle** [comment]
OK I finally managed to get a Windows build with AppVeyor.  It's now uploaded to the latest release.  Would you please give that a try when you get a chance?

**Comment by wereturtle** [comment]
Yes I think it had to do with the way the characters were being counted from the bytestream read from cmark-gfm's API.  I had to make some adjustments in how I invoked the API.
